include/threads/MutexManager.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <mutex>
+#include <unordered_map>
+#include <vector>
+
+namespace threads {
+
+class MutexManager
+{
+  public:
+    void createMutex(int mutexId);
+
+    void lockMutex(int mutexId);
+
+    void unlockMutex(int mutexId);
+
+    bool tryLockMutex(int mutexId);
+
+    void destroyMutex(int mutexId);
```

Comment:
This is a new class to handle shared mutexes between modules. It's really just a map and wrapper around the existing `std::thread` but it means the implementation can be shared by all types of `WasmModule` and avoids nesting this complexity in the intrinsic threading functions.